Lib/test/test_doctest/test_doctest.py

```diff
@@ -18,8 +18,12 @@
 import contextlib
 
 
-if not support.has_subprocess_support:
-    raise unittest.SkipTest("test_CLI requires subprocess support.")
+def doctest_skip_if(condition):
+    def decorator(func):
+        if condition:
```

Comment:
```suggestion
        if condition and support.HAVE_DOCSTRINGS:
```

There's no need to modify docstrings when running in optimized mode.